packages/core/core/src/TransformerRunner.js

```diff
@@ -136,28 +111,27 @@ class TransformerRunner {
       }
 
       // If the generated asset has the same type as the input...
-      if (result.type === context.type) {
+      // TODO: this is incorrect since multiple file types could map to the same pipeline. need to compare the pipelines.
+      if (result.type === input.type) {
         // If we have reached the last transform in the pipeline, then we are done.
         if (pipeline.length === 1) {
-          assets.push(
-            asset || (await transformerResultToAsset(input, result, context))
-          );
+          assets.push(await finalize(asset, generate));
         } else {
           // Recursively run the remaining transforms in the pipeline.
-          let nextInput = transformerResultToInput(input, result);
+          let nextInput = input.createChildAsset(result);
           let cacheEntry = await this.runPipeline(
             nextInput,
             pipeline.slice(1),
             null,
-            getNextContext(context, result)
+            generate
           );
 
           assets = assets.concat(cacheEntry.assets);
           connectedFiles = connectedFiles.concat(cacheEntry.connectedFiles);
```

Comment:
I got a little confused here because I thought that cacheEntry was only passed in to the first call of `runPipeline`. I guess it has part of the shape of a CacheEntry, but it might help to not give it the same name as a function parameter. Maybe `nextPipelineResult`?

Reply:
👍 originally it was a full cache entry and then I changed it